File: RB101/small_problems/easy_4/short_long_short.rb
# Write a method that takes two strings as arguments, determines the longest of the two strings,
# and then returns the result of concatenating the shorter string, the longer string,
# and the shorter string once again. You may assume that the strings are of different lengths.

# Input: Two different strings
# Output: A new string that is the result of concatenating the shorter string in the front, the long string in the middle, and the short string again at the end

# Rules
# Assume we're getting two strings, don't validate
# The long word has to be sandwiched in the middle.
# You can assume one will be equal (they're never equal)

# Algo
# Use a simple if statement to check one length against the other
# Return concatenated string that direction
# Else, return it the other direction

def short_long_short(string1, string2)
  if string1.length > string2.length
    long = string1
    short = string2
  else
    long = string2
    short = string1
  end
  "#{short}#{long}#{short}"
end

# Test cases
p short_long_short('abc', 'defgh') == "abcdefghabc"
p short_long_short('abcde', 'fgh') == "fghabcdefgh"
p short_long_short('', 'xyz') == "xyz"

File: PY101/small_problems/easy_1/short_long_short.py
# Write a function that takes two strings as arguments, determines the length
# of the two strings, and then returns the result of concatenating the shorter
# string, the longer string, and the shorter string once again. You may assume
# that the strings are of different lengths.

def short_long_short(text1, text2):
    if len(text2) < len(text1):
        short = text2
        long = text1
    elif len(text1) < len(text2):
        short = text1
        long = text2
    else:
        print('These strings are the same length. Try again.')
        return None

    return f'{short}{long}{short}'

# These examples should all print True
print(short_long_short('abc', 'defgh') == "abcdefghabc")
print(short_long_short('abcde', 'fgh') == "fghabcdefgh")
print(short_long_short('', 'xyz') == "xyz")
print(short_long_short('xyz', 'xyz'))